### PR TITLE
Sprint/filedicts/resolve file movements

### PR DESCRIPTION
**Pull request by denizural**
New functionality `resolve_file_movements` that may replace `assemble` in the future

### CODE REVIEW

File: tests/test_esm_runscripts/test_filedicts.py [hunk 1]
```diff
@@ -12,12 +12,20 @@
...
 
 import esm_runscripts.filedicts
+from esm_runscripts.filedicts import resolve_file_movements
+
+
+@pytest.fixture()
+def config():
+    """Generates fake config to be used before each test"""
+    fake_config = dict()
+    yield fake_config
```
Comment (pgierz) [inline tests/test_esm_runscripts/test_filedicts.py:29]: Why does this need to be a yield? (Just out of curiosity)

File: tests/test_esm_runscripts/test_filedicts.py [hunk 2]
```diff
@@ -12,12 +12,20 @@
   unit test, which you don't really want.  
 """
 import os
+from pathlib import Path
 
 import yaml
-
-from pathlib import Path
+import pytest
```
Comment (pgierz) [inline tests/test_esm_runscripts/test_filedicts.py:18]: The import order might change once in a while. I typically run `isort` from my vim whenever a Python file is saved, but that is a trivial change that might happen now and then.

File: tests/test_esm_runscripts/test_filedicts.py [hunk 3]
```diff
@@ -12,12 +12,20 @@
...
 
 import esm_runscripts.filedicts
+from esm_runscripts.filedicts import resolve_file_movements
```
Comment (pgierz) [inline tests/test_esm_runscripts/test_filedicts.py]: Could be removed, you already have the entire module imported above.

File: tests/test_esm_runscripts/test_filedicts.py [hunk 4]
```diff
@@ -95,3 +103,11 @@ def test_cp(fs):
     esm_runscripts.filedicts.SimulationFile.cp(source, target)
 
     assert os.path.exists(target)
+
+
+def test_resolve_file_movements(config):
+    # act
+    config = resolve_file_movements(config)
```
Comment (pgierz) [inline tests/test_esm_runscripts/test_filedicts.py]: I would rather call the function via `module.func_name`, as it is done in other tests.

**Review by pgierz — CHANGES_REQUESTED** [review]
Looks good. I would remove the `filelists.py` from this PR to keep it focused on only new functionality.

File: tests/test_esm_runscripts/test_filedicts.py (hunk 4 above)
Comment (mandresm) [reply tests/test_esm_runscripts/test_filedicts.py]: agreed

**Review by mandresm — COMMENTED** [review]
Everything looks good to me too after resolving the requests from @pgierz. I would also remove the changes of `filelists.py` from this PR and use them for another one, maybe even for `release`.

File: tests/test_esm_runscripts/test_filedicts.py (hunk 1 above)
Comment (denizural) [reply tests/test_esm_runscripts/test_filedicts.py:29]: `yield` is recommended by pytest: https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#yield-fixtures-recommended

File: tests/test_esm_runscripts/test_filedicts.py (hunk 4 above)
Comment (denizural) [reply tests/test_esm_runscripts/test_filedicts.py]: What about this? `import esm_runscripts.filedicts as filedicts`

**Review by mandresm — APPROVED** [review]

**Comment by mandresm** [comment]
Thanks!

**Pull request merged**